game.go

```diff
@@ -301,17 +297,24 @@ func instance(gamer reflect.Value) *Game {
 	return fld.Addr().Interface().(*Game)
 }
 
-func getFieldPtrOrAlloc(v reflect.Value, i int) (name string, val interface{}) {
+func getFieldPtrOrAlloc(g *Game, v reflect.Value, i int) (name string, val interface{}) {
 	tFld := v.Type().Field(i)
 	vFld := v.Field(i)
 	typ := tFld.Type
 	word := unsafe.Pointer(vFld.Addr().Pointer())
 	ret := reflect.NewAt(typ, word).Interface()
-	if vFld.Kind() == reflect.Ptr && typ.Implements(tySpriter) {
+
+	if vFld.Kind() == reflect.Ptr && typ.Implements(tySprite) {
 		obj := reflect.New(typ.Elem())
 		reflect.ValueOf(ret).Elem().Set(obj)
 		ret = obj.Interface()
 	}
+
+	if vFld.Kind() == reflect.Interface && typ.Implements(tySprite) && g.canBindSprite(tFld.Name) {
```

Comment:
It is strange to check `canBindSprite` here (in `getFieldPtrOrAlloc`).

If you want to make sure you can get the type from `g.typs`, maybe it's better to check "if `tFld.Name` exists in `g.typs`"?

Reply:
done